cloudanalysis.fd/NonVarCldLib/cloud_saturation.f90

```diff
@@ -291,37 +294,43 @@ function ruc_saturation(Temp,pressure)
 !
 !_____________________________________________________________________
 
-  use constants, only: rd_over_cp, h1000,one,zero
+  use constants, only: one,zero
   use kinds, only: r_single,i_kind, r_kind
 !
     implicit none
     real(r_single) :: ruc_saturation
+    real           :: RSLF,RSIF
 
     REAL(r_kind),  intent(in) :: Temp       ! temperature in K
     real(r_single),intent(in) :: pressure   ! pressure  (hpa)
+    real :: temp4,press4
 
-    real(r_kind) ::  es0_p
-    parameter (es0_p=6.1121_r_kind)     ! saturation vapor pressure (mb)
-    real(r_kind) SVP1,SVP2,SVP3
-    data SVP1,SVP2,SVP3/es0_p,17.67_r_kind,29.65_r_kind/
+!ruc    real(r_kind) ::  es0_p
+!ruc    parameter (es0_p=6.1121_r_kind)     ! saturation vapor pressure (mb)
+!ruc    real(r_kind) SVP1,SVP2,SVP3
+!ruc    data SVP1,SVP2,SVP3/es0_p,17.67_r_kind,29.65_r_kind/
 
     real(r_kind) :: temp_qvis1, temp_qvis2
     data temp_qvis1, temp_qvis2 /268.15_r_kind, 263.15_r_kind/
 
-    REAL(r_kind) :: evs, qvs1, eis, qvi1, watwgt
+!ruc    REAL(r_kind) :: evs, qvs1, eis, qvi1, watwgt
+    REAL(r_kind) :: qvs1, qvi1, watwgt
 !
-
+     temp4=Temp            ! convert to real
+     press4=pressure*100.0 ! covert to Pa
 !
 ! evs, eis in mb
 !   For this part, must use the water/ice saturation as f(temperature)
-        evs = svp1*exp(SVP2*(Temp-273.15_r_kind)/(Temp-SVP3))
-        qvs1 = 0.62198_r_kind*evs/(pressure-evs)  ! qvs1 is mixing ratio kg/kg
+!ruc        evs = svp1*exp(SVP2*(Temp-273.15_r_kind)/(Temp-SVP3))
+!ruc        qvs1 = 0.62198_r_kind*evs/(pressure-evs)  ! qvs1 is mixing ratio kg/kg
+        qvs1 = RSLF(press4,temp4)
```

Comment:
Maybe something similar to this: qvs1 = RSLF(pressure_pa, real(Temp, kind=r_kind))